config/initializers/yaml_renderer.rb

```diff
@@ -1,4 +1,7 @@
 ActionController::Renderers.add :yaml do |obj, _|
-  data = JSON.load(obj.to_json).to_yaml
+  # Create a serializable resource instance
+  serializable_resource = ActiveModelSerializers::SerializableResource.new(obj)
+
+  data = serializable_resource.as_json.to_yaml
```

Comment:
Sorry, I forgot to mention `serializable_hash`. I had tried it as well and it gives symbols as keys:
```Ruby
data = ActiveModelSerializers::SerializableResource.new(user)
data = data.serializable_hash
 => {:id=>1, :handle=>"sonalkr132", :email=>"aditya.prakash132@gmail.com"}
data = YAML.dump(data)
 => "---\n:id: 1\n:handle: sonalkr132\n:email: aditya.prakash132@gmail.com\n"
```